examples/godocbot/pkg/controller/pullrequest/godoc_deployer.go

```diff
@@ -0,0 +1,265 @@
+		// update the status
+		prCopy := pr.DeepCopy()
+		prCopy.Status.GoDocLink = fmt.Sprintf("https://%s.serveo.net/pkg/%s/%s/%s", prinfo.subdomain(), prinfo.host, prinfo.org, prinfo.repo)
+		if err = r.Client.Update(ctx, prCopy); err != nil {
+			return reconcile.Result{}, err
+		}
+		log.Printf("godoc link updated successfully for pr %v", request.NamespacedName)
+	}
+	return reconcile.Result{}, nil
+}
+
+// deploymentForPullRequest creates a deployment object for a given PullRequest.
+func deploymentForPullRequest(pr *v1alpha1.PullRequest) (*appsv1.Deployment, error) {
+	// we are good with running with one replica
+	var replicas int32 = 1
+
+	prinfo, err := parsePullRequestURL(pr.Spec.URL)
+	if err != nil {
+		return nil, err
+	}
+	prinfo.commitID = pr.Spec.CommitID
+
+	labels := map[string]string{
+		"org":  prinfo.org,
+		"repo": prinfo.repo,
+	}
+
+	tunnelArgs := "80:localhost:6060"
+	tunnelArgs = fmt.Sprintf("%s:%s", prinfo.subdomain(), tunnelArgs)
+
+	dep := &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "apps/v1",
+			Kind:       "Deployment",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      pr.Name,
+			Namespace: pr.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(pr, schema.GroupVersionKind{
+					Group:   v1alpha1.SchemeGroupVersion.Group,
+					Version: v1alpha1.SchemeGroupVersion.Version,
+					Kind:    "PullRequest",
+				}),
+			},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Image:           "gcr.io/sunilarora-sandbox/godoc:0.0.1",
+							Name:            "godoc",
+							ImagePullPolicy: "Always",
+							Command:         []string{"/bin/bash"},
+							Args:            prinfo.godocContainerArgs(),
+						},
+						{
+							Image:           "gcr.io/sunilarora-sandbox/ssh-client:0.0.2",
+							Name:            "ssh",
+							ImagePullPolicy: "Always",
+							Command:         []string{"ssh"},
+							Args:            []string{"-tt", "-o", "StrictHostKeyChecking=no", "-R", tunnelArgs, "serveo.net"},
+						},
+					},
+				},
+			},
+		},
+	}
+	return dep, nil
+}
+
+// prInfo is an structure to represent PullRequest info. It will be used
+// internally to more as an convenience for passing it around.
+type prInfo struct {
+	host     string
+	org      string
+	repo     string
+	pr       int64
+	commitID string
+}
+
+// parsePullRequestURL parses given PullRequest URL into prInfo instance.
+// An example pull request URL looks like:
+// https://github.com/kubernetes-sigs/controller-runtime/pull/15
+func parsePullRequestURL(prURL string) (*prInfo, error) {
+	u, err := url.Parse(prURL)
+	if err != nil {
+		return nil, err
+	}
+	parts := strings.Split(strings.TrimPrefix(u.Path, "/"), "/")
```

Comment:
consider path.Split

Reply:
path.Split gives `dir` and `file` which is different use case.